tiledb/api/src/array/dimension.rs

```diff
@@ -110,18 +160,80 @@ impl<'ctx> Dimension<'ctx> {
 impl<'ctx> Debug for Dimension<'ctx> {
     fn fmt(&self, f: &mut Formatter) -> FmtResult {
         let json = json!({
+            "name": self.name(),
             "datatype": format!("{}", self.datatype()),
+            "cell_val_num": self.cell_val_num(),
             "domain": fn_typed!(self.domain, self.datatype() => match domain {
-                Ok(x) => format!("{:?}", x),
-                Err(e) => format!("<{}>", e)
+                Ok(x) => json!(x),
+                Err(e) => json!(e)
             }),
-            /* TODO: filters */
+            "extent": fn_typed!(self.extent, self.datatype() => match extent {
+                Ok(x) => json!(x),
+                Err(e) => json!(e),
+            }),
+            "filters": format!("{:?}", self.filters()),
             "raw": format!("{:p}", *self.raw)
         });
         write!(f, "{}", json)
     }
 }
 
+impl<'c1, 'c2> PartialEq<Dimension<'c2>> for Dimension<'c1> {
```

Comment:
There is a proptest for reflexivity but no test that changing fields changes the result.  I need to add that.